runner/runner/handlers.py

```diff
@@ -35,22 +37,30 @@ def run_task(_=None, task=None):
     return {
         "task_id": task["id"],
         "status": exit_status,
-        "output": output.decode(),
-        "result": result.decode(),
+        "output": output.decode() if isinstance(output, bytes) else output,
+        "result": result.decode() if isinstance(result, bytes) else result,
     }
 
 
 def _run_task(task):
     package = task.get("package")
     function = task.get("function")
     parameters = json.dumps(task["function_parameters"])
+    variables = task.get("variables")
     run_command = ["--function", function, "--parameters", parameters]
 
     logging.info("Running %s from package %s", function, package)
     docker_client = docker.from_env()
-    container = docker_client.containers.run(
-        package, auto_remove=False, detach=True, command=run_command
-    )
+    try:
+        container = docker_client.containers.run(
+            package,
+            auto_remove=False,
+            detach=True,
+            command=run_command,
+            environment=variables,
+        )
+    except DockerException:
+        return (1, "Unable to execute function", "")
```

Comment:
Had to add this because I keep making a `PATH` variable and that blows away the containers `PATH`. This results in the container failing to start and you get 0 feedback anywhere. It's not the best for now, but it at least doesn't leave the task in an incorrect state.

Reply:
This seems fine.  I would include the exception message in your output though since I'm guessing the additional context would be helpful:

```python
except DockerException as exc:
    return (1, f"Unable to execute function. Encountered error: {exc}", "null")
```

Also the "null" instead of empty string because that's what would normally come back if nothing is returned.